query/sources/impl/src/main/java/org/codice/ddf/admin/sources/csw/CswSourceUtils.java

```diff
@@ -85,23 +91,52 @@ public CswSourceUtils(ConfiguratorSuite configuratorSuite) {
      * Attempts to discover the source from the given hostname and port with optional basic authentication.
      * <p>
      * Possible Error Codes to be returned
-     * - {@link org.codice.ddf.admin.common.report.message.DefaultMessages#CANNOT_CONNECT}
+     * - {@link org.codice.ddf.admin.common.report.message.DefaultMessages#UNKNOWN_ENDPOINT}
      *
      * @param hostField address to probe for CSW capabilities
      * @param creds     optional credentials for basic authentication
-     * @return a {@link ReportWithResultImpl} containing the {@link UrlField} or an {@link org.codice.ddf.admin.api.report.ErrorMessage} on failure.
+     * @return a {@link ReportWithResultImpl} containing the {@link CswSourceConfigurationField} or an {@link org.codice.ddf.admin.api.report.ErrorMessage} on failure.
      */
-    public ReportWithResult<ResponseField> discoverCswUrl(HostField hostField,
+    public ReportWithResult<CswSourceConfigurationField> getConfigFromHost(HostField hostField,
             CredentialsField creds) {
-        return requestUtils.discoverUrlFromHost(hostField,
-                URL_FORMATS,
-                creds,
-                GET_CAPABILITIES_PARAMS);
+        List<List<SourceTaskCallable<CswSourceConfigurationField>>> taskList = new ArrayList<>();
+
+        for (List<String> urlFormats : URL_FORMATS) {
+            List<SourceTaskCallable<CswSourceConfigurationField>> callables = new ArrayList<>();
+            urlFormats.forEach(url -> callables.add(new SourceTaskCallable<>(url,
```

Comment:
✏️ Instead of modifying the list with the `forEach`, try
```java
List<SourceTaskCallable<CswSourceConfigurationField>> callables = urlFormats.stream()
        .map(url -> new SourceTaskCallable<>(url,
                hostField,
                creds,
                this::getCswConfigFromUrl))
        .collect(Collectors.toList());
```